woocommerce/woocommerce-ios
# [Order creation] Fix/allow taps on order form notices

## Description


While working on #10986, I noticed that tapping the buttons on the notices from the Order Form didn't work. This was due to a gesture recogniser blocking taps to the button.

This PR moves the gesture recogniser to the view, and improves the Notice wrapper.

## Testing instructions


1. Set a breakpoint in Proxyman or Charles for POST responses to `https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/*/rest-api/`, but leave it disabled.
2. Launch the app and navigate to the Orders tab
3. Tap Create
4. Tap Add Product
5. Select a product
6. Enable your breakpoint
7. Tap Done in the app to add your product
8. Edit the response to an error code, e.g. 500 Internal Server Error
9. Observe that the retry notice is shown
10. Tap the button
11. Observe that the request is retried.


## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/2472348/2c9273de-1691-43dc-91bb-05d5f15b9507

---
